### PR TITLE
fix: using deeplinks

### PR DESCRIPTION
**Pull request by ignaciosantise**
## Summary
* Changed redirect logic to use deeplinks as first option
* Added fallback to universal link in case the app is not installed
* Showing error if deeplink & universal link fails

#### Installed app

https://github.com/WalletConnect/web3modal-react-native/assets/25931366/e105fafa-cdec-42e1-a739-576da84951f0

#### Not installed app

https://github.com/WalletConnect/web3modal-react-native/assets/25931366/982d6d99-9371-4a18-9593-fc086ffbd0f3

#### Not installed app with bad link

https://github.com/WalletConnect/web3modal-react-native/assets/25931366/21d08302-ae76-4c4a-8110-09762232a4a9

#### Installed app

https://github.com/WalletConnect/web3modal-react-native/assets/25931366/90136999-f61d-454b-b419-ede62b544d49

#### Not installed app

https://github.com/WalletConnect/web3modal-react-native/assets/25931366/31c4af07-3d8d-4906-bfaf-d68ac45fa0b3

#### Not installed app with bad link

https://github.com/WalletConnect/web3modal-react-native/assets/25931366/512bdf24-3ce4-4f47-9be1-cab6206e58fe

### CODE REVIEW

File: example/src/screens/App.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import '../../expo-crypto-shim.js';
-import { Alert, SafeAreaView, StyleSheet } from 'react-native';
+// import '../../expo-crypto-shim.js';
```
Comment (0xAsimetriq) [inline example/src/screens/App.tsx:1]: Is this comment still needed? if so perhaps add a `TODO`

File: src/utils/CoreUtil.ts [hunk 2]
```diff
@@ -3,7 +3,9 @@ export const CoreUtil = {
     return url.startsWith('http://') || url.startsWith('https://');
   },
 
-  formatNativeUrl(appUrl: string, wcUri: string): string {
+  formatNativeUrl(appUrl: string, wcUri: string): string | undefined {
+    if (!appUrl || appUrl === '') return undefined;
```
Comment (0xAsimetriq) [inline src/utils/CoreUtil.ts:7]: `!appUrl` handles both cases here I think?

**Review by 0xAsimetriq — APPROVED** [review]
Left few comments 👍

File: example/src/screens/App.tsx (hunk 1 above)
Comment (ignaciosantise) [reply example/src/screens/App.tsx:1]: Thanks for pointing this! Apparently expo now works without that shim 🎉  i'll remove it

**Pull request merged**